docs/semgrep-code/semgrep-assistant-code.md

```diff
@@ -54,14 +54,49 @@ Semgrep Assistant supports all the same languages as Semgrep Code. See [Supporte
 * For more details, see the [Semgrep Assistant FAQ](https://get.semgrep.dev/assistant).
 
+## Required GitHub or GitLab repository permissions
+
+Semgrep Assistant requires the same permissions that Semgrep needs to integrate with GitHub or GitLab such as permissions that allow Semgrep to run CI jobs and post comments to PRs or MRs. See [<i class="fa-regular fa-file-lines"></i> Requested permissions for GitHub and GitLab](/semgrep-cloud-platform/getting-started/#requested-permissions-for-github-and-gitlab) for a list of permissions.
+
+Assistant extends normal Semgrep functionality by providing contextually aware AI-generated suggestions. In order to build that context, it requires extra permissions in GitHub and GitLab.
+
+<Tabs
+    defaultValue="github"
+    values={[
+    {label: 'GitHub', value: 'github'},
+    {label: 'GitLab', value: 'gitlab'},
+    ]}
+>
+
+<TabItem value='github'>
+
+Semgrep Assistant requires [read access to your code in GitHub](https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28). This is done through installation of a private Semgrep GitHub app that you install during Assistant setup. This private Semgrep GitHub app:
+
+* Is fully under your control so you can revoke access or specific permissions at any time by visiting **Settings > Applications** in GitHub.
+* Only accesses source code repositories on a file-by-file basis; it does not need or request org-level access to your codebase.
+* Can be configured to limit its scope to specific repositories. You do not need to give read access to all repositories in your GitHub organization. 
+
+</TabItem>
+
+<TabItem value='gitlab'>
+
+Semgrep Assistant requires the **API scope** to run in GitLab or self-managed GitLab. This can be specified at either the [project access token level](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html) or [personal access token level](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html). 
+
+* You can revoke [project level access tokens](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html#revoke-a-project-access-token) or [personal access tokens](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#revoke-a-personal-access-token) at any time. 
+* Semgrep Assistant only accesses source code repositories on a file-by-file basis; it does not need or request org-level access to your codebase.
+* The token can be configured to limit its scope to specific projects or individuals. You do not need to give read access to all repositories in your GitLab organization).
```

Comment:
Should we use the term "project" throughout, here, rather than switching to "repositories"?

Reply:
I can do that. :)